fieldmanager.php

```diff
@@ -435,6 +443,100 @@ function fm_trigger_context_action() {
 }
 add_action( 'init', 'fm_trigger_context_action', 99 );
 
+/**
+ * Fire an action for the current Fieldmanager context for the REST API.
+ *
+ * This is separate from fm_trigger_context_action since rest_pre_dispatch fires on parse_request
+ * and is too late to be part of the existing context hooks.
+ *
+ * @since 1.3.0
+ *
+ * @param mixed           $result  Response to replace the requested version with. Can be anything
+ *                                 a normal endpoint can return, or null to not hijack the request.
+ * @param WP_REST_Server  $server  Server instance.
+ * @param WP_REST_Request $request Request used to generate the response.
+ */
+function fm_trigger_rest_context_action( $result, $server, $request ) {
+	// Get post type and taxonomy endpoints available for the REST API.
+	$post_type_endpoints = fm_get_registered_object_rest_base( 'post_types' );
+	$taxonomy_endpoints = fm_get_registered_object_rest_base( 'taxonomies' );
+
+	// Get the route for the request for comparison.
+	$route = $request->get_route();
+
+	// If the route is empty, we cannot continue.
+	if ( empty( $route ) ) {
+		return;
+	}
+
+	// Store matches to use with the context action.
+	$matches = array();
+
+	// Use regexes to find the right context.
+	if (
+		! empty( $post_type_endpoints )
+		&& preg_match( '#/wp/v2/(' . implode( '|', array_keys( $post_type_endpoints ) ) . ')(/?)(.*?)/#i', $route, $matches )
+	) {
+		// Post Types.
+		do_action( "fm_post_{$post_type_endpoints[ $matches[1] ]}" );
+
+		// Quickedit.
+		do_action( "fm_quickedit_{$post_type_endpoints[ $matches[1] ]}" );
```

Comment:
While firing `fm_quickedit_*` addresses the particular issue described earlier, it potentially introduces a separate backwards-compatibility concern.

Is there any precedent in FM for firing multiple context hooks during the same request? Existing FM code using the context actions might already assume that `fm_quickedit_*` will fire on only `edit.php` and not concurrently with `fm_post_*`, for example.

A broader backwards-compatibility concern is that `fm_get_context()` and `fm_calculate_context()` don't work as expected during REST API requests.

Reply:
> A broader backwards-compatibility concern is that fm_get_context() and fm_calculate_context() don't work as expected during REST API requests.

Yeah, I think having those two functions handle the action hooks for all REST API events is the best solution.  This work was taken from a branch that Brad worked on and since it worked I just kept it as is.  Maybe we should start to think about how we can leverage `fm_get_context()` and `fm_calculate_context()` to call the proper hooks during a REST request.